rioxarray/rioxarray.py

```diff
@@ -372,6 +376,13 @@ def set_crs(
         :obj:`xarray.Dataset` | :obj:`xarray.DataArray`:
             Dataset with crs attribute.
         """
+        warnings.warn(
```

Comment:
I think raising a warning would be fine. However, `rio.set_crs` is used internally elsewhere. To avoid raising warnings where it is not needed, I suggest renaming this to `rio._set_crs` and use it internally. Then, create a new `rio.set_crs` that calls `rio._set_crs` after raising a warning.

Reply:
Good to know. I'll look into it.